pkg/providers/docker/docker.go

```diff
@@ -226,6 +227,17 @@ func buildTemplateMapMD(clusterSpec *cluster.Spec) map[string]interface{} {
 		"eksaSystemNamespace": constants.EksaSystemNamespace,
 		"kubeletExtraArgs":    kubeletExtraArgs.ToPartialYaml(),
 	}
+
+	if len(clusterSpec.Spec.WorkerNodeGroupConfigurations[0].Taints) > 0 {
+		var taints []corev1.Taint
+		for _, taint := range clusterSpec.Spec.WorkerNodeGroupConfigurations[0].Taints {
+			if taint.Effect != "NoExecute" && taint.Effect != "NoSchedule" {
+				taints = append(taints, taint)
+			}
+		}
+		values["workerNodeGroupTaints"] = taints
```

Comment:
Can we just have something like the below to move the rest of this logic to `spec.go` which is provider agnostic?

```
values["workerNodeGroupTaints"] = clusterSpec.WorkerNodeGroupTaints()
```

Reply:
We will check during verification.